### PR TITLE
Add standard aliases for continue, step, next, finish commands

### PR DESCRIPTION
**Pull request by iogakos**
Natively support aliases for standard ruby-debug commands

```
`c` for `continue`
`s` for `step`
`n` for `next`
`f` for `finish`
```

I thought it would be convenient for the above aliases to be natively supported instead of maintaining and moving around the corresponding configuration section of ~/.pryrc.

Thanks

--j

### CODE REVIEW

**Comment by banister** [comment]
Thanks :) however `pry-debugger` originally supported these aliases but they were removed as they conflict with local variables that have the same name. Sorry.

**Comment by iogakos** [comment]
Hm, I see. But doesn't this argument stands for every alias set by `pry-debugger`? 

```
   => 22:     n = 10
      23:     binding.pry
      24:  end

pry(#<Foo>)> n

      22:     n = 10
   => 23:     binding.pry
      24:  end

pry(#<Foo>)> p n
10
pry(#<Foo>)>
```

I find the above a reasonable workaround when aliases conflict with local variables. Thanks for your time anyway.

--j

**Comment by banister** [comment]
I agree with you generally, but i think the issue was that variables such as `n` and `s` are very commonly used scratch local variables so the chance of conflict is especially high.

Another option for you is to define function-key based (e.g `f5` or `f6`) aliases for `next` and `step` in your `.inputrc`. Perhaps @rking could give you some pointers on how to do that :)

**Comment by rking** [comment]
gem install pry-fkeys

Then run `inputrc?` to read the explanation then run `inputrc!` to doctor up your `~/.inputrc` file for these changes.

If you're running the crummy Readline implementation, it will tell you so and also tell you the steps neeeded ot fix that.

**Comment by rking** [comment]
@jgakos Un-pro move Tweeting about this when you don't seem to understand the issue. =P

**Comment by iogakos** [comment]
TL;DR;
To whom this notification may not concern, don't read this if you don't have time to waste. --j

I am not exactly sure what is pro, so I can hardly determine the definition of un-pro. But, turning a pull request into a personal confrontation, I don't think lies in the first of the above terms (pro). Understanding the issue is not our issue here, we are talking about plain simple, childish aliases and aliases are made to make developers' life easier, not harder. So, there is nothing to understand.

This is my approach

_ I used your code as an **enhanced** wrapper of `ruby-debug`.
_ Surprisingly noticed you have removed the most convenient aliases that `ruby-debug` supports.
_ Publicly expressed my frustration on this (something you apparently call `unpro`)
_ Forked your repo, wrote a patch, opened a pull request, got rejected, showed there is a clean way to override the reason I got rejected, rejected again, thanked you.
_ Got characterized as `unpro`.

_I won't say nothing about your proposal installing another gem for supporting a standard alias with another key which resides on the top of my keyboard_

It is simple as that. I don't agree using non-descriptive variables like `n` or `f` or `c`. I don't agree with you removing functionality of the gem you are trying to enhance (`debugger`) at all. You were supposed to enhance its functionality, not remove any of its features. This is where we disagree and it is fair enough.

_edit: respecting people of the community watching the gem, next involvement in this conversation by me will be only on the actual reason of this PR's existence_

Thanks ~~for maintaining the gem.~~

--j

**Comment by banister** [comment]
@jgakos rking is not a maintainer of the `pry-debugger` gem.

**Comment by rking** [comment]
The fundamental difference of pry vs debugger is that pry executes input as Ruby, since pry is a REPL first and a debugger second.

This means that if you added commands that took precedence over Ruby code, you could easily violate the Principle of Least Surprise. Imagine if a user had this code:

``` ruby
def nuke_dir n
  binding.pry; system 'rm', '-rf', "../../../#{n}"
end
```

Now, let's say they wanted to see the value of `n`, because it was sometimes `nil`. They'll examine it in pry by typing `n<enter>`, which if your patch was accepted, would merrily, recursively, remove all of `../../..`.

---

Regarding the "Keys which reside at the top of your keyboard", if I understand correctly you're saying it's less ergonomic.  Actually, with the `n/s/c/f` aliases, they're really `n<enter>/s<enter>/c<enter>/f<enter>`, so it's more work even though it's on the main tray of keys. I think you'd like the F-keys approach if you gave it a bit of time.

**Comment by banister** [comment]
@jgakos would you be happy with `%n` and `%s` aliases? If these are ok, you can resubmit your pull request and i'll merge it :)

**Comment by iogakos** [comment]
@banister hm, I don't think it will actually solve the issue.

Thanks

--j

**Comment by deivid-rodriguez** [comment]
Hi, I like this PR. I will be merging it into pry-byebug.

Salud!

**Comment by deivid-rodriguez** [comment]
@banister, @rking Hi again, I merged this into pry-byebug a while ago but it doesn't feel quite right actually. I might revert that.

I would like instead to add the aliases only if I'm starting pry through `binding.pry` inside the code but don't add them if I'm starting pry from the command line through `pry`, `rails c` or whatever. Is there anyway, via configuration files or not, to do this?

Thanks a lot.

**Comment by banister** [comment]
Yeah,

you can define a `when_started`  hook, and check whether the binding is NOT `Pry.toplevel_binding`

I guess something like this:

``` ruby
Pry.config.hooks.add_hook(:when_started, :define_byebug_aliases) do |target, _, _|
  define_the_aliases if target != Pry.toplevel_binding
end
```

Try something like that, if it doesnt work, ping me again and i'll help you with it.

_EDIT:_ missed a not ;)

**Comment by deivid-rodriguez** [comment]
Woow, thanks a lot! I'll let you if I need more help. I think this might be useful for people who normally use pry as a debugger when starting it with `binding.pry` but normally use it as REPL when starting it from the outset.

Congratulations for such a nice gem by the way.
